src/operations/delete.ts

```diff
@@ -65,7 +65,7 @@ export class DeleteOneOperation extends CommandOperation<DeleteOptions, DeleteRe
   execute(server: Server, callback: Callback<DeleteResult>): void {
     const coll = this.collection;
     const filter = this.filter;
-    const options = this.options;
+    const options = Object.assign({}, this.options, this.bsonOptions);
```

Comment:
I won't comment on every instance to spare you the emails, but I think we should prefer the spread operator to assign: `const options = {...this.options, ...this.bsonOptions}`. But not a strong opinion, and spread doesn't work in every case, like when assigning onto `'this`